accepted/targeting-packs/targeting-packs.md

```diff
@@ -0,0 +1,149 @@
+  Identifier as .NET Framework) using the .NET CLI
+
+## Requirements
+
+### Goals
+
+* Doesn't require running an installer
+* Doesn't require Windows
+* Can be used in SDK-style multi-targeting projects
+
+### Non-Goals
+
+* Supporting NuGet-based acquisition of the .NET Framework runtime
+* Supporting NuGet-based acquisition from non-SDK-style projects
+* Support pre-.NET Framework 4.5 targeting packs
+* Support non-.NET Framework targeting packs (Xamarin, Mono, Unity)
+
+## Design
+
+* Daniel has created a [prototype](https://github.com/dsplaisted/ReferenceAssemblyPackages).
+    - We have an identity package
+    - Each version has a different identity and is looped in by TFM
+    - Ensures only incremental download
+    - Open question is whether we need the aggregate package at all or just
+      compute the package ID from the TFM, but that's not what we did for
+      `Microsoft.NetCore.App` or `NETStandard.Library`. What was the reason?
+
+* Includes .targets files
+    - Locate the files.
+    - Design time?
+
+### Details
+
+* There will be separate NuGet packages with reference assemblies for each
+  version of .NET Framework. This means that projects targeting a single version
+  of .NET Framework don't need to download and spend disk space on the reference
+  assemblies and intellisense files for all the other possible versions of .NET
+  Framework. It also means that when a new version of .NET Framework is
+  released, the reference assemblies for the previous versions don't need to be
+  re-shipped in an updated package along with the new assemblies.
+* There will be a single "metapackage" that can be referenced, that will have
+  conditional dependencies on each of the version-specific reference assembly
+  packages.
+* The metapackage will automatically be referenced by the .NET SDK when required
+  (ie on non-Windows OS's). On Windows, we will probably make the .NET SDK
+  continue to rely on the reference assemblies from the targeting packs by
+  default.
+* For the package IDs, I suggest the following:
+  - For the metapackage: `Microsoft.NETFramework.ReferenceAssemblies`
+  - For the version-specific packages:
+    `Microsoft.NETFramework.ReferencesAssemblies.net462` (where net462 is
+    replaced with the corresponding NuGet short framework Identifier)
+* The NuGet packages will include a `.targets` file that sets the
+  `TargetFrameworkRootPath` to a path within the NuGet package. This will allow
+  the existing MSBuild logic in the `GetReferenceAssemblyPaths` target and other
+  logic such as automatically referencing the facades if necessary to continue
+  to work as normal.
+* The logic in the `.targets` file in the NuGet packages will only set the
+  `TargetFrameworkRootPath` if the `TargetFrameworkIdentifier` and
+  `TargetFrameworkVersion` properties match the reference assemblies that the
+  package provides. It will also only set the property if the
+  `UseReferenceAssembliesFromPackage` property is set to true.
```

Comment:
I'm having second thoughts on requiring opt-in via a `UseReferenceAssembliesFromPackage` property if you've already referenced the package.  I don't think it's necessary and just complicates using the packages from non-SDK projects.  SDK projects would have a different property to control whether the packages are referenced at all.

Reply:
If someone explicitly adds the package reference I agreed we don't need this opt-in. However if the package reference is implicitly added by the SDK then I do think we need some opt-in/out but that could be around the implicit package reference in the SDK though and not in the targets.

Reply:
@dsplaisted, @nguerrera and myself talked yesterday:

* We concluded that the opt-in for non-SDK-style project is the manual install of the package. If the package is present, it will prefer the reference assemblies from it even if a TP is installed.
* For SDK-style projects we concluded that it's OK to default to use the package and use the existing opt-out mechanism via "use implicit package references" (not sure what the property is, but we have one)

I'll update the proposal to capture this later.

Reply:
The property to disable implicit package references in the SDK is called `DisableImplicitFrameworkReferences`.